regolith/validators.py

```diff
@@ -109,4 +113,49 @@ def ensure_email(email):
     'databases': (always_false, ensure_databases),
     'stores': (always_false, ensure_stores),
     'email': (always_false, ensure_email),
-    }
+}
+
+
+# TODO: run this as a lint like thing rather than raising at the first error
+def validate_schema(record, schema, keys=()):
+    """Validate a database record against a schema
+
+    Parameters
+    ----------
+    record : dict
+        The database record to be tracked
+    schema : dict
+        The schema to validate the record against
+    keys : tuple
+        The sequential keys for the data being accessed, used for
+        printing/debuging errors
```

Comment:
Broadly speaking, I guess I'd prefer if this returned (bool, str message) rather than raising an error.  That way the call site can decide whether or not to raise the error. Also, it allows the message to be appended to so that all of the ways that the record doesn't adhere to the schema can be known with one call to this function.

Reply:
I think this is done with cerberus. (validator.errors)